### DISCUSSION TITLE
Provide iCal link

### DISCUSSION THREAD

**Discussion by JackBailey** [hargata/lubelog#697] [Ideas]
I think it'd be great if it was possible for lubelog to provide an ical link, which would allow other calendar apps to subscribe to it, so you can easily see upcoming reminders on your phone without opening the website.

**Comment by JoramHhub** [comment]
This would be so convenient!

**Comment by hargata** [comment]
we've spent some time looking into this, we can provide an API method that returns a ICS file, but the biggest challenge is security since most calendar apps don't really support authorization when it comes to a simple calendar subscription.

**Reply by JackBailey** [reply]
Ah that makes sense - Personally that'd work fine for me though, as my instance isn't exposed. 

Although, it then obviously wouldn't work with calendars you don't host yourself, so maybe a base64encoded login as a URL query parameter might work in lieu of something like an API key.

**Reply by hargata** [reply]
Got it figured out, turns out most calendar apps DO support basic auth, tho it is a legacy spec that is subject to deprecation eventually.

Next release, for instances that have auth configured, you can use:

`https://username:password@mylubeloggerdomain/api/calendar`

where username and password are URL encoded(e.g.: `1234!` will become `1234%21`

For instances without auth configured, you can use:

`https://mylubeloggerdomain/api/calendar`

**Reply by a-hersa** [reply]
Fantastic news. I have limited knowledge but other services I use that integrate through iCal just share a link where anyone with the link can have access. I see the `username:password` option feature useful when there is more than one user per instance. Will the `https://mylubeloggerdomain/api/calendar` share the admin calendar or..?

**Reply by hargata** [reply]
if there's no auth enabled, meaning only one user exist for the entire instance, then yeah, `https://mylubeloggerdomain/api/calendar` will return all reminders for all vehicles.

**Comment by a-hersa** [comment]
I was searching how to implement this when I saw this post and that it's already in the backlog. Thank you so much.

**Comment by hargata** [comment]
Released.

Use the following format:

`https://username:password@lubeloggerdomain/api/calendar`
